server/request_handler.go

```diff
@@ -117,14 +126,17 @@ func (r *RpcRequestHandler) processBatchRequest(client RPCProxyClient, jsonBatch
 	for i := 0; i < cap(resCh); i++ {
 		// Process each individual request
 		// Scatter worker
-		go func(count int, rpcReq *types.JsonRpcRequest) {
+		go func(count int, rpcReq *types.JsonRpcRequest, record *requestRecord) {
+			id := uuid.New()
 			// Create child logger
-			logger := log.New(log.Ctx{"uid": fmt.Sprintf("%s.%d", r.uid, count)})
+			logger := log.New(log.Ctx{"uid": r.uid, "id": id, "count": count})
+
+			entry := r.requestRecord.UpdateEthSendRawTxEntries(rpcReq)
```

Comment:
in there, the `RequestId` should already be set, then you don't need to pass it into `NewRpcRequest`

Reply:
Also I think `UpdateEthSendRawTxEntries` does too much magic. I think more easily readable would be to have the check if this method is eth_sendRawTx here on the outside, and rename `UpdateEthSendRawTxEntries` to `AddEthSendRawTxEntry`

Reply:
updated